File: sprint5/By languages/C#/lesson3code1.cs
Node FindNode(Node root, int value) {
  if (root == null) {
    return null;
  }
  if (value < root.Value) {
    return FindNode(root.Left, value);
  }
  if (value == root.Value) {
    return root;
  }
  if (value > root.Value) {
    return FindNode(root.Right, value);
  }
}

File: sprint5/By languages/swift/lesson3code1.swift
func findNode<T: Comparable>(_ root: Node<T>?, _ value: T) -> Node<T>? {
    // Если мы пришли в поддерево, а его не существует, значит, нужного элемента в дереве поиска нет
    guard let rootNode = root else {
        return nil
    }
    
    if value < rootNode.value {
        return findNode(rootNode.left, value)
    } else if value == rootNode.value {
        return rootNode
    } else {
        return findNode(rootNode.right, value)
    }
}
